client/lib/signup/step-actions/index.js

```diff
@@ -315,21 +315,24 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 }
 
 export function setThemeOnSite( callback, { siteSlug, themeSlugWithRepo, selectedDesign } ) {
```

Comment:
Rather than overloading this action function, I'm wondering whether we should have a new `setDesignOnSite` or `setThemeAndDesignOnSite` function instead. That way each step action only does one thing, rather than having one function that can operate in multiple modes.